typescript/src/generator.ts

```diff
@@ -65,22 +71,38 @@ export class Generator {
     }
   }
 
-  private async readGrpcServiceConfig(parameter: string) {
-    const match = parameter.match(/^["']?grpc-service-config=([^"]+)["']?$/);
-    if (!match) {
-      throw new Error(`Parameter ${parameter} was not recognized.`);
+  private getParamMap(parameter: string) {
+    // Example: "grpc-service-config=texamplejson","package-name=packageName"
+    const paramMap: Map = {};
+    const parameters = parameter.split(',');
+    for (let param of parameters) {
+      // remove double quote
+      param = param.substring(1, param.length - 1);
+      const arr = param.split('=');
+      paramMap[arr[0].toKebabCase()] = arr[1];
     }
-    const filename = match[1];
-    if (!fs.existsSync(filename)) {
-      throw new Error(`File ${filename} cannot be opened.`);
+    return paramMap;
+  }
+
+  private async readGrpcServiceConfig(map: Map) {
```

Comment:
It makes sense to make this map a class member, and just access `this.map`.

Reply:
sure!